### PR TITLE
HeaderPropagation: reworded registration exception for clarity

### PR DESCRIPTION
**Pull request by alefranz**
The MessageHandler throws when used outside of a request.
This reword the exception to clarify that this could be another possible cause of it.

Also fixes a missing `'` in the current message.

Related to #12169, but not a final fix

@rynowak I have a couple questions:
- What is the best way to describe an incoming request of the pipeline? Open for suggestions on how to better word this message.
- I guess this is it for now. We could use the `HttpContextAccessor` to distinguish between the two types of errors and have a specific message, but this can add a performance hit if not already registered. Unless we use it only if it is available and fallback to this generic exception message otherwise. However adding a dependency only to have a better error message feels wrong.

/cc @anurse

### CODE REVIEW

File: src/Middleware/HeaderPropagation/src/HeaderPropagationMessageHandler.cs [hunk 1]
```diff
@@ -48,8 +48,8 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
             {
                 var message =
                     $"The {nameof(HeaderPropagationValues)}.{nameof(HeaderPropagationValues.Headers)} property has not been " +
-                    $"initialized. Register the header propagation middleware by adding 'app.{nameof(HeaderPropagationApplicationBuilderExtensions.UseHeaderPropagation)}() " +
-                    $"in the 'Configure(...)' method.";
+                    $"initialized. Register the header propagation middleware by adding 'app.{nameof(HeaderPropagationApplicationBuilderExtensions.UseHeaderPropagation)}()' " +
+                    $"in the 'Configure(...)' method. Also, do not use an HttpClient with header propagation outside of an incoming HTTP request.";
```
Comment (anurse) [inline src/Middleware/HeaderPropagation/src/HeaderPropagationMessageHandler.cs]:
```suggestion
                    $"in the 'Configure(...)' method. Header propagation can only be used within the context of an HTTP request.";
```

**Review by analogrelay — COMMENTED** [review]
Wordsmithing! Would like @rynowak to confirm all is in the right place (and my words are good), but otherwise LGTM

File: src/Middleware/HeaderPropagation/src/HeaderPropagationMessageHandler.cs (hunk 1 above)
Comment (alefranz) [reply src/Middleware/HeaderPropagation/src/HeaderPropagationMessageHandler.cs]: Updated

**Review by analogrelay — APPROVED** [review]
Looks good! I'll take a look at the CI failures. They may not be related.

**Comment by anurse** [comment]
Yeah, looks like a timeout. Re-running.

**Comment by alefranz** [comment]
CI looks good now

**Review by rynowak — APPROVED** [review]

**Comment by rynowak** [comment]
Thanks @alefranz

**Pull request merged**